pkg/hub/submarineragent/controller.go

```diff
@@ -588,21 +576,20 @@ func (c *submarinerAgentController) cleanUpSubmarinerClusterEnv(ctx context.Cont
 	}
 
 	managedClusterInfo := config.Status.ManagedClusterInfo
-	cloudProvider, err := cloud.GetCloudProvider(c.restConfig, c.kubeClient, c.manifestWorkClient, c.dynamicClient,
+	cloudProvider, err := cloud.GetCloudProvider(nil, c.kubeClient, c.manifestWorkClient, c.dynamicClient,
 		nil, c.eventRecorder, managedClusterInfo, config)
 	if err != nil {
 		//TODO handle the error gracefully in the future
 		c.eventRecorder.Warningf("CleanUpSubmarinerClusterEnvFailed", "failed to create cloud provider: %v", err)
```

Comment:
Incase you happen to respin the PR. 

"failed to get cloud provider"

Reply:
done